src/app/components/chat/client.ts

```diff
@@ -161,20 +166,54 @@ function reset(chat: ChatClient) {
 	chat.messageQueue = [];
 }
 
+export async function setGuestToken(chat: ChatClient, guestToken: string) {
+	const tokenChanged = chat.guestToken !== guestToken;
+	chat.guestToken = guestToken;
+
+	console.log('isGuest: ' + chat.isGuest);
+	console.log('tokenChanged: ' + tokenChanged);
+
+	if (!chat.isGuest || tokenChanged) {
+		destroy(chat);
```

Comment:
Yeah, this is because i have to set the isGuest bit between the destroy and connect in this case because destroy unsets it.

I wonder if its cleaner to just not reset isGuest? we're also not resetting guest token so maybe this is actually preferable..